jlh/JonFirstApp/app/src/main/java/com/e_conomic/jonfirstapp/MainActivity.java

```diff
@@ -1,68 +1,146 @@
+        }
 
-        // If orientation is in landscape add message to fragment.
-        if (getResources().getConfiguration().orientation ==
-                Configuration.ORIENTATION_LANDSCAPE) {
+    }
 
-            // Get the fragment that shows the message.
-            if (display_message_fragment == null) {
-                FragmentManager fragmentManager = getSupportFragmentManager();
-                display_message_fragment =
-                        fragmentManager.findFragmentById(R.id.displaymessage_fragment);
-            }
+    /** Called when the user clicks the show/hide button */
+    public void showHide(View view) {
 
-            // Edit the text view (show written message).
-            TextView textView = (TextView)
-                    display_message_fragment.getView().findViewById(R.id.display_message);
-            textView.setText(message);
+        // If the displaying fragment has not been initialized return.
+        if (display_message_fragment == null) {
+            return;
+        }
 
-        // If orientation is in portrait start new activity.
-        } else {
-            // Create intent.
-            Intent intent = new Intent(this, DisplayMessageActivity.class);
+        // Setup FragmentManager and FragmentTransaction.
+        View fragment_container = findViewById(R.id.fragment_container);
+        FragmentManager fragmentManager = getSupportFragmentManager();
+        FragmentTransaction ft = fragmentManager.beginTransaction();
+        ft.setCustomAnimations(android.R.anim.fade_in, android.R.anim.fade_out);
+
+
+        // Hide Fragment if it is visible.
+        if (display_message_fragment.isVisible()) {
+            ft.hide(display_message_fragment).commitNow();
+
+            isDisplayMessageFragmentVisible = false;
+
+            // Inform the main fragment of the change
+            main_fragment.setButtonText();
+
+            // Set width of fragment layout container to 0.
+            changeFragmentLayout(fragment_container, 0);
+            return;
+        }
+
+        // Show Fragment if it is hidden.
+        if (display_message_fragment.isHidden()) {
+            ft.show(display_message_fragment).commitNow();
+
+            isDisplayMessageFragmentVisible = true;
 
-            // Add (user) entered message to the created intent.
-            intent.putExtra(EXTRA_MESSAGE, message);
+            // Inform the main fragment of the change
+            main_fragment.setButtonText();
 
-            // Start an instance of the DisplayMessageActivity specified by the intent.
-            startActivity(intent);
+            // Set width of fragment layout container to fill half the screen.
+            changeFragmentLayout(fragment_container, ViewGroup.LayoutParams.MATCH_PARENT);
+            return;
         }
 
+    }
+
+    /** Helper function to change the layout of the fragment container */
+    private void changeFragmentLayout(View fragment_layout, int newLayoutParam) {
```

Comment:
Please tell me there's another way to hide and show a fragment (**and** dynamically change the view of the fragment besides) than what implemented in this method? The hide and show functions I used doesn't quite seem to do the trick. For instance, when on landscape, it leaves the entire right side white, instead of "stretching" the main fragment to fill the entire screen 😨